app/app_xdg.go

```diff
@@ -119,9 +126,15 @@ func rootConfigDir() string {
 }
 
 func watchTheme() {
-	go portalSettings.OnSignalSettingChanged(func(changed portalSettings.Changed) {
-		if changed.Namespace == "org.freedesktop.appearance" && changed.Key == "color-scheme" {
-			fyne.CurrentApp().Settings().(*settings).setupTheme()
-		}
-	})
+	go func() {
+		// with portal this may not be immediate, so we update a cache instead
+		currentVariant.Store(findFreedesktopColorScheme())
```

Comment:
Similar suggestion here:
```suggestion
		currentVariant.Store(fyne.ThemeVariant(findFreedesktopColorScheme()))
```